### ISSUE TITLE
Missing overload for two-argument form of `.get` for `TypedDict` with nested `TypedDict`

### ISSUE THREAD

**Issue by shiftinv** [microsoft/pyright#2898]
**Describe the bug**
First of all, I'm not entirely sure if this should be considered a bug or a feature request; maybe both. It depends, I suppose the current behavior could be considered working as intended.

The `.get` method of a `TypedDict` with an element like `inner: XYZ` seems to be missing an overload like `(k: Literal['inner'], default: XYZ) -> XYZ`. This generally isn't an issue, except for `TypedDict`s (and implicitly cast types in general?).

**To Reproduce**
```py
from typing import TypedDict

class Nested(TypedDict):
    number: int

class Container(TypedDict):
    inner: Nested


d: Container = ...  # type: ignore

reveal_type(d.get("inner", {"number": 3}))  # Type of "d.get("inner", { "number": 3 })" is "Nested | dict[str, int]"
```

**Expected behavior**
```py
Type of "d.get("inner", { "number": 3 })" is "Nested"
```

**VS Code extension or command-line**
pyright 1.1.212

**Additional context**
This behavior seems to have changed some time after 1.1.208, as pylance currently matches the expected behavior.
Technically pyright .212 isn't wrong, which is why I suppose this could be considered a feature request as well; it could be nice if pyright resolved the return value to `"Nested"` instead of `"Nested | dict[str, int]"`, if possible.

Using `reveal_type(d.get)` shows this:
```py
# pyright 1.1.212
Type of "d.get" is "Overload[(k: Literal['inner']) -> Nested | None, (k: Literal['inner'], default: Unknown) -> (Nested | Unknown), (k: str) -> Any | None, (k: str, default: Unknown) -> (Any | Unknown)]"
# pyright 1.1.208 / pylance
Type of "d.get" is "Overload[(k: Literal['inner'], default: Nested = ...) -> Nested, (k: str) -> Any | None, (k: str, default: Any = ...) -> Any]"
```

**Comment by erictraut** [comment]
Yeah, that's a reasonable request. The previous behavior was wrong because it required that the second argument be the same as the field's value type. So previously pyright synthesized the following overloads:

```python
    @overload
    def get(self, key: Literal["<key>"]) -> value_type | None: ...

    @overload
    def get(self, key: Literal["<key>"], default: value_type) -> value_type: ...
```

As of pyright 1.1.210, this was changed to:
```python
    @overload
    def get(self, key: Literal["<key>"]) -> value_type | None: ...

    @overload
    def get(self, key: Literal["<key>"], default: _T) -> value_type | _T: ...
```

As you point out, it's probably better to include a combination of these three overloads:
```python
    @overload
    def get(self, key: Literal["<key>"]) -> value_type | None: ...

    @overload
    def get(self, key: Literal["<key>"], default: value_type) -> value_type: ...

    @overload
    def get(self, key: Literal["<key>"], default: _T) -> value_type | _T: ...
```

I'll make that change for the next release.

**Comment by GBeauregard** [comment]
Oh, I think this resolves also the Sequence annoyance from before because of the ordering; I don't know why I didn't think of just trying both overloads then.

**Comment by erictraut** [comment]
@GBeauregard, yes, that's a good point. It should address that issue too.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.213, which I just published. It will also be included in the next release of pylance.
